framework/doc/content/source/meshgenerators/XYZMeshPlaneCutter.md

```diff
@@ -0,0 +1,68 @@
+# XYZMeshPlaneCutter
+
+!syntax description /Mesh/XYZMeshPlaneCutter
+
+## Overview
+
+The `XYZMeshPlaneCutter` is basically the 3D version of [`XYMeshLineCutter`](/XYMeshLineCutter.md). It is used to slice a 3D input mesh along a given plane, and discard the portion of the mesh on one side of the plane. The input mesh, given by [!param](/Mesh/XYZMeshPlaneCutter/input), must be 3D and contain only first-order elements. The cutting plane is specified by [!param](/Mesh/XYZMeshPlaneCutter/plane_normal) and [!param](/Mesh/XYZMeshPlaneCutter/plane_point), which are two `libMesh::Point` type data that represent the normal vector of the cutting plane and a point on the cutting plane, respectively. This mesh generator removes the part of the mesh located on the side of the plane in the direction of the normal vector. The mesh is then smoothed to ensure a straight cut instead of a "zigzag" cut along element boundaries as generated by [`PlaneDeletionGenerator`](/PlaneDeletionGenerator.md).
+
+## Methods
+
+`XYZMeshPlaneCutter` first converts all elements of the input mesh into `TET4` elements. Next, the `TET4` elements sliced by the cutting plane are further split into `TET4` elements. 
```

Comment:
@miaoyinb can you please check whether the comments @roystgnr made on #25433 apply to the method you are using to go from hexes to tets?

we should consolidate with #25433 imo here
the 3D cutter could call the conversion generator as a sub-generator.

Reply:
I will take a look

Reply:
I looked into #25433. I did make some efforts to make sure an element is split consistently with its neighbors. Not sure what the current status of #25433 is. I could convert my current PR into one conversion (splitting) MG and one cutting MG if it works with everyone. @roystgnr @aprilnovak

Reply:
> I could convert my current PR into one conversion (splitting) MG and one cutting MG if it works with everyone

works for me. I think it s a good idea. I agree let's coordinate on this, select a conversion method for hex -> tet that is as robust as possible.

Reply:
I tried @roystgnr 's suggestions to stress-test the implementation in #25433, but I could not come up with a scenario where the mesh failed to work for a heat conduction test case. So I am kind of stuck on how to address Roy's comments.

Reply:
When I was developing this MG, I found if a QUAD face shared by two HEX8 elements is not split consistently when splitting the two HEX8 elements, the output mesh still SEEMS to work fine for a simple diffusion problem. But I guess it would not work properly if the physics becomes more complex?

Reply:
It might not error, but the result would be wrong. 
Ideally we would add a diagnostic to the meshDiagnosticGenerator to detect these situations.

Reply:
IIRC, I found that the result was identical -- I used a postprocessor to compute the L2 norm against a known method of manufacturing solutions result.

Reply:
And the solution was non trivial (wasn't just luck) right? Not just a linear solution to a source less diffusion problem for example. 

The two elements on each side of the former quad face would not be neighbors to each other.
That would definitely be a problem for some physics (finite volume and ray tracing ones at least). I could see how linear Lagrange functions could still look alike because the nodal values are the ones imposing the continuity

Reply:
Correct, it was a non-trivial solution. The MMS solution I chose was `exp(x)+exp(y)+z*z`.

Reply:
It's not impossible that you'd still get convergence with a broken Tet4 topology tested with an H1 problem; there aren't actually any interior degrees of freedom there to go completely loose; you'd still have some kind of control over all four vertices on the broken face.  But that wouldn't be success, it would be misleading luck.  A Discontinuous Galerkin problem would fail, because it wouldn't see neighbors across the broken face and would miss the jump terms.  Do one level of h refinement after the split and it would fail, because there'd be nothing constraining (at least one of) the new elements' vertices to each other.  Turn the Tet4 elements into Tet10 and it would fail, because there'd be nothing constraining one of the new edge nodes to the perpendicular edge node in the same place.  I'm not even sure you'd even expect to get convergence in general as opposed to in heat conduction specifically; there might be something that works with the symmetry of Poisson but breaks with more complicated physics.  The very first thing you do to derive a Continuous Galerkin method is integrating by parts, with continuity assumptions that aren't true if you add a discontinuity in the middle of your functions.

> select a conversion method for hex -> tet that is as robust as possible.

"as robust as possible" would be "just call all_tri() in libMesh".  But that's also 4x as expensive as the orientation-dependent splitting, creating 24 tets instead of 6 in a Hex for example.  There's something to be said for the less robust option when it works; we just need to make sure on any given mesh that it's actually working and not just kinda-working.

I still think my suggestion in https://github.com/idaholab/moose/pull/25433#issuecomment-1761958018 was reasonable: make sure we added exactly as many neighbor links as we expected to; if we got fewer then there was a break somewhere.

A more sophisticated option would be to check on each square face rather than globally: create a `map<array<dof_id_type,4>,dof_id_type>`.  Every time you split the square connecting points with ids N,M,P,O, you look up `your_map[{M,N,O,P}]` (sorting for uniqueness), and if it's not already there you set it to the lowest id of the diagonal you're splitting on, and if it is there you check that it already is set to the lowest id of the diagonal you're splitting on.

Reply:
For the element splitting algorithm I used in this PR, the quad face splitting is always consistent. In some cases the created TET4 elements might have poor quality (which also implies that the original HEX8/PRISM6/PYRAMID5 element might also have quality issue) .

If we choose to use this algorithm, do we still need a checker described by @roystgnr to double-check after the splitting ? Thanks.

Reply:
I would say no we dont need it here because we expect the splitting to be consistent.

However I would still have conversion and cutting in two separate mesh generators.
And in conversion you should plan to offer multiple "conversion techniques"

Reply:
Another consistent criteria with possibly better quality would be to split the quad faces in a way that maximizes the quality of the triangles

In case of a draw use the id of the nodes to lift the indetermination

This won't necessarily get used the best quality tets but it would maybe be enough to avoid the worst split decisions

Reply:
The method I use now to split a quad face is to select the diagonal line that involves the node that has the minimal global node id. Using this approach, the six faces of a HEX8 element are not independent. That is good because it ensures that the HEX8 element can be split into 6 TET4 elements without need for adding additional node(s).

The method you mention here seems to make the split of these quad faces independent from each other (maybe there is still some interdependence but weaker than the first approach). That is, not all of the possible combinations can lead to a TET4 splitting without need for add node(s). It is likely that we will need to create another node inside the HEX8 to split it into 12 TET4 elements in some cases.

Reply:
Good point.

Reply:
> The method I use now to split a quad face is to select the diagonal line that involves the node that has the minimal global node id.

This is clever.  I'd still like to see some level of debugging assertions (i.e. not the current level of 0; this is not trivial code), ideally applied to unit tests on permuted element node orderings, but in theory if it works it should eliminate the need to do post-facto checks in optimized runs.

IMHO I wouldn't worry about tet quality.  Low-order tets have to be slivers or pancakes to really hurt your approximation space constants, but I believe the only way to get a pancake or a sliver out of a 6-tet decomposition of a hex is if the hex was already of *much* worse quality.

Reply:
I am working on the TET conversion MG based on my algorithm here. It seems better to move the methods to a mesh utility compared to the subgenerator option as we do not need to convert the part of the mesh that will be removed.

Reply:
Utility is good imo. 
The "conversion" generator will leverage the utility